src/services/CweService.ts

```diff
@@ -0,0 +1,50 @@
+import { osimRuntime } from '@/stores/osimRuntime';
+
+const DATA_KEY = 'CWE:API-DATA';
+const VERSION_KEY = 'CWE:API-VERSION';
+
+export async function fetchAndCacheAPI() {
+  if (!osimRuntime.value.backends.mitre) {
+    console.log('No Mitre backed configured, skipping CWE API cache update.');
+    return;
+  }
+  const baseUrl = osimRuntime.value.backends.mitre;
+  try {
+    const [versionResponse] = await Promise.all([
+      fetch(`${baseUrl}/cwe/version`),
+    ]);
+
+    if (!versionResponse.ok) {
+      throw new Error('Failed to fetch CWE version');
+    }
+
+    const versionData = await versionResponse.json();
+    const storedVersion = localStorage.getItem(VERSION_KEY);
+
+    if (storedVersion !== versionData.ContentVersion) {
+      localStorage.setItem(VERSION_KEY, versionData.ContentVersion);
```

Comment:
I would move this lower, if the data fetching fails, we will have the new version string but the old data and it won't be updated until a new version is released.
Fetch everything and if nothing fails store the data and then the version